hazelcast/src/main/java/com/hazelcast/aggregation/impl/BigDecimalAverageAggregator.java

```diff
@@ -57,4 +62,27 @@ public BigDecimal aggregate() {
                 BigDecimal.valueOf(count));
     }
 
+    @Override
+    public int getFactoryId() {
+        return AggregatorDataSerializerHook.F_ID;
+    }
+
+    @Override
+    public int getId() {
+        return AggregatorDataSerializerHook.BIG_DECIMAL_AVG;
+    }
+
+    @Override
+    public void writeData(ObjectDataOutput out) throws IOException {
+        out.writeUTF(attributePath);
+        out.writeObject(sum);
```

Comment:
Yes, this big decimal seems like problem for non-java clients. 
Or may be non-java client dont even need to support this class ever ?